airflow/utils/log/s3_task_handler.py

```diff
@@ -84,7 +85,10 @@ def close(self):
             # read log and remove old logs to get just the latest additions
             with open(local_loc, 'r') as logfile:
                 log = logfile.read()
-            self.s3_write(log, remote_loc)
+            successfully_written = self.s3_write(log, remote_loc)
+            # If the file was written out, remove it to avoid duplicating logs in S3
+            if successfully_written:
```

Comment:
Since this PR modifies behavior on which some users may rely, I would suggest adding a configuration option to explicitly enable deletion on upload.

Reply:
Agreed. You can default it to true.

Also this should be mentioned in UPDATING.md

Reply:
Ok, I'll make those changes. 

Because this bug should also pop-up for users backing up logs into Azure/GCS, I'm going to go ahead and add this behavior for those log file handlers as well.

Reply:
Hey friends, changes made. Any other changes desired?